### PR TITLE
Add cli entrypoint for compile

### PR DESCRIPTION
**Pull request by GeorgesLorre**
WIP!

First implementation of the fondant CLI (tried and tested):

`fondant compile -p /playground/mycoolpipeline.py:pipeline`

(note you will need to (re)run `poetry install --all-extras`)

I'm mainly looking for feedback on this approach before I spend more time expanding it

### CODE REVIEW

File: fondant/cli.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import argparse
+import importlib.util
+import re
+
+from fondant.compiler import DockerCompiler
+from fondant.pipeline import Pipeline
+
+
+class PipelinePythonReference(argparse.Action):
```
Comment (GeorgesLorre) [inline fondant/cli.py:9]: I did not find anything standard on importstrings (`file.py:my_object`) so I made my own parser.

Comment (RobbeSneyders) [reply fondant/cli.py:9]: Any reason to go for a custom `Action` instead of a custom `type`? I think a custom `type` could be a bit simpler.

FYI, [reference implementation](https://github.com/encode/uvicorn/blob/master/uvicorn/importer.py#L9) from `uvicorn`.

File: fondant/cli.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            return match.groups()
+        else:
+            raise ValueError(f"Could not parse input {input}")
+
+    @staticmethod
+    def load_pipeline(module: str, object: str) -> Pipeline:
+        """Attempt to load a pipeline from a module file path and object name."""
+        spec = importlib.util.spec_from_file_location(object, module)
+        pipeline = importlib.util.module_from_spec(spec)  # type: ignore
+        spec.loader.exec_module(pipeline)  # type: ignore
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply fondant/cli.py:42]: See my comment above for an implementation based on `import_module` and `getattr`.

File: examples/pipelines/starcoder/pipeline.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 """Pipeline used to create the dataset to train the StarCoder model."""
 
+import argparse
```
Comment (RobbeSneyders) [inline examples/pipelines/starcoder/pipeline.py:3]: Why was this added? I think we should be able to remove the `if __name__ = "__main__":` block at the bottom of the file, no?

File: fondant/compiler.py [hunk 4]
```diff
@@ -129,6 +129,11 @@ def _generate_spec(self, pipeline: Pipeline, extra_volumes: list) -> dict:
                 ]
             )
 
+            # add component spec to command if does not exist yet
+            if not component_op.arguments.get("component_spec"):
+                component_spec = component_op.component_spec.specification
+                command.extend(["--component_spec", json.dumps(component_spec)])
+
```
Comment (RobbeSneyders) [inline fondant/compiler.py:136]: You added this to the `ComponentOp` class in #215, right?

**Review by RobbeSneyders — APPROVED** [review]
Thanks @GeorgesLorre! I know it must hurt to see the coverage decline, but looking at the uncovered lines, I don't think there's a lot to test without just duplicating the same logic.

File: examples/pipelines/starcoder/docker-compose.yml
Comment (RobbeSneyders) [inline examples/pipelines/starcoder/docker-compose.yml:1]: This should be left out of git.

**Review by RobbeSneyders — CHANGES_REQUESTED** [review]
One more thing :wink:

**Review by RobbeSneyders — APPROVED** [review]
Thanks!
